src/state/CallViewModel.ts

```diff
@@ -1112,6 +1145,128 @@ export class CallViewModel extends ViewModel {
     this.scope.state(),
   );
 
+  private readonly handsRaisedSubject = new BehaviorSubject<
+    { userId: string; date: Date | null }[]
+  >([]);
+  private readonly reactionsSubject = new BehaviorSubject<
+    { userId: string; reaction: ReactionOption; ttl: number }[]
+  >([]);
+
+  public addHandRaised(userId: string, date: Date) {
+    this.handsRaisedSubject.next([{ userId, date }]);
+  }
+
+  public removeHandRaised(userId: string, date: Date | null) {
+    this.handsRaisedSubject.next([{ userId, date }]);
+  }
+
+  public addReaction(userId: string, reaction: ReactionOption, ttl: number) {
+    this.reactionsSubject.next([{ userId, reaction, ttl }]);
+  }
+
```

Comment:
This has been my approach so far where the VM has some subjects, and the useReactions provider has been updating these subjects with the latest values. The dependent components then subscribe to the relevant observable, and all is well.

The hook is still used to to *send* reactions / hand raised at the moment.